Kanan/CookingMod.cpp

```diff
@@ -16,25 +16,35 @@ namespace kanan {
-        g_cooking->m_hook->callOriginal(_this, 1, cookingTwo);
-        g_cooking->m_hook->callOriginal(_this, 2, cookingThree);
+        auto ingredientIdx = _this[193]; //Interface button we are currently clicking
+        if (ingredientIdx > -1 && ingredientIdx < 3) { //-1 is a state of not clicking any button. 0 through 2 are each Add button starting left
+            //Manually set each food percent of the object from our kanan UI
+                _this[188] = cookingOne;//+2F0
+                _this[189] = cookingTwo;
+                _this[190] = cookingThree;
 
-        //work around if you add some ingredient before enabled cooking mod, it may mess up amounts.
-        _this[188] = cookingOne;
-        _this[189] = cookingTwo;
-        _this[190] = cookingThree;
+                //subtract 1 from the button we are pressing to allow the the original function to set the cooking bar full flag correctly
+                //If we don't do this the code thast handles activating the start button does not get executed
+                _this[188 + ingredientIdx] = _this[188 + ingredientIdx] - 1;
+                //we call the original function - effecdtively adding the 1 value we just subtracted
+                g_cooking->m_hook->callOriginal(_this, amount);
+                return;
+        } else {
+            g_cooking->m_hook->callOriginal(_this, amount);
+            return;
+        }
     }  
 
     //find the addresses we need for patching
     CookingMod::CookingMod():m_hook(nullptr) {
-        auto cookingAddress = scan("client.exe", "40 57 48 83 EC 30 48 C7 44 24 20 FE FF FF FF 48 89 5C 24 40 48 89 74 24 48 48 8B F1 44 8B 89 F8 02 00 00 44 03 89 F4 02 00 00 44 03 89 F0 02 00 00 41 81 F9 40 42 0F 00 ");
-       
+    auto cookingAddress = scan("client.exe", "48 89 ? ? ? 48 89 ? ? ? 56 57 41 ? 48 83 EC ? 0F 29 ? ? ? 44 8B ? 48 8B ? 45 33 ? 0F B6 ? ? ? ? ? E8 ? ? ? ? 44 8B ? ? ? ? ? 41 83 F9 ? 0F 84 ? ? ? ? 41 83 F9 ? 0F 8D ? ? ? ? 8B 85 ? ? ? ? 85 C0 0F 84 ? ? ? ? 44 3B ? 0F 86 ? ? ? ? 41 8B ? 2B C8 0F AF ? ? ? ? ? B8 ? ? ? ? F7 E1 C1 EA ? 44 8B ? ? ? ? ? 44 03 ? ? ? ? ? 44 03 ? ? ? ? ? 41 81 F8 ? ? ? ? 0F 83 ? ? ? ? 41 8D ? ? B9 ? ? ? ? 41 2B ? 3D ? ? ? ? 0F 46 ? 85 C9 0F 84 ? ? ? ? 42 01 ? ? ? ? ? ? 48 8B ? E8 ? ? ? ? 84 C0 0F 84 ? ? ? ? 48 8B ? ? ? ? ? E8 ? ? ? ? 48 8B ? E8 ? ? ? ? 48 8B ? 48 C7 44 24 70 ? ? ? ? 48 8D ? ? ? ? ? E8 ? ? ? ? 8B F8 8D 48 ? E8 ? ? ? ? 48 89 ? ? ? 48 8D ? ? 4C 8D ? ? ? ? ? 8D 57 ? E8 ? ? ? ? 48 8B ? ? ? 89 79 ? 41 B0 ? 48 8D ? ? ? 48 8B ? E8 ? ? ? ? 90 48 8B ? ? ? 48 85 ? 74 ? E8 ? ? ? ? 48 C7 44 24 70 ? ? ? ? 41 B8 ? ? ? ? 48 8D ? ? ? ? ? 48 8B ? ? ? ? ? E8 ? ? ? ? 48 8B ? ? ? ? ? 48 8B ? B2 ? FF 90 ? ? ? ? 44 89 ? ? ? ? ? 48 8B ? ? ? ? ? 48 8B ? ? ? ? ? E8 ? ? ? ? 8B BD ? ? ? ? 03 BD ? ? ? ? 03 BD ? ? ? ? 48 85 ? 0F 84 ? ? ? ? 8D 4F ? 81 F9 ? ? ? ? 0F 87 ? ? ? ? 48 8B ? ? ? ? ? 48 8B ? E8 ? ? ? ? 48 8B ? E8 ? ? ? ? 0F 28 ? 48 8B ? E8 ? ? ? ? 0F 28 ? F3 0F ? ? ? ? ? ? 48 8B");
```

Comment:
I'm a bit creeped out by this huge pattern. When I get some time I may revisit this pattern myself and look for ways of adding some sophistication to how this address is found.

Reply:
No resolution necessary.

Reply:
Just to clarify - what was the reason for not needing a resolution?